esphome/components/deep_sleep/deep_sleep_esp32.cpp

```diff
@@ -13,6 +13,7 @@ optional<uint32_t> DeepSleepComponent::get_run_duration_() const {
     switch (wakeup_cause) {
       case ESP_SLEEP_WAKEUP_EXT0:
       case ESP_SLEEP_WAKEUP_EXT1:
+        return this->wakeup_cause_to_run_duration_->ext1_cause;
```

Comment:
```suggestion
#if defined(USE_ESP32_VARIANT_ESP32H2)
        return this->wakeup_cause_to_run_duration_->ext1_cause;
#endif
```
I believe that without this it would be breaking change.